Tasks/TwineAuthenticateV1/twineauthenticatemain.ts

```diff
@@ -47,14 +41,58 @@ async function main(): Promise<void> {
         let pypircPath = utils.getPypircPath();
 
         // create new file. We do not merge existing files and always create a fresh file
-        fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
-        tl.setVariable("PYPIRC_PATH", pypircPath, false);
-        tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        if (!tl.getVariable("PYPIRC_PATH") || !tl.exist(tl.getVariable("PYPIRC_PATH"))) {
+            fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
+            tl.setVariable("PYPIRC_PATH", pypircPath, false);
+            tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        }
+        else {
+            const pypirc = fs.readFileSync(tl.getVariable("PYPIRC_PATH"), 'utf8');
+            let fileContent = ini.parse(pypirc);
+
+            let dotRepos: string[] = [];
+
+            for (let entry of newEndpointsToAdd) {
+
+                tl.debug(tl.loc("Info_AddingAuthForRegistry", entry.packageSource.feedName));
+
+                if (entry.packageSource.feedName in fileContent){
+                    tl.debug(tl.loc("DuplicateRegistry", entry.packageSource.feedName));
+                    if (internalFeed.includes(entry)) {
+                        internalFeed.pop();
+                        continue;
+                    }
+                    externalEndpoints.pop();
+                    continue;
+                }
+
+                if (entry.packageSource.feedName.includes('.')) {
+                    let repo = `${os.EOL}[${entry.packageSource.feedName}]${os.EOL}repository=${entry.packageSource.feedUri}${os.EOL}username=${entry.username}${os.EOL}password=${entry.password}`;
+                    dotRepos.push(repo);
+                }
+                else {
+                    fileContent[entry.packageSource.feedName] = new Repository(
+                        entry.packageSource.feedUri,
+                        entry.username,
+                        entry.password
+                    );
+                }
+
+                fileContent["distutils"]["index-servers"] += " " + entry.packageSource.feedName;
+            }
+
+            let encodedStr = ini.encode(fileContent);
+            fs.writeFileSync(pypircPath, encodedStr);
+
+            for (let repo of dotRepos) {
+                fs.appendFileSync(tl.getVariable("PYPIRC_PATH"), repo, 'utf8');
+            }
+        }
 
         // Configuring the pypirc file
         internalFeedSuccessCount = internalFeed.length;
         externalFeedSuccessCount = externalEndpoints.length;
-        console.log(tl.loc("Info_SuccessAddingAuth", internalFeedSuccessCount, externalFeedSuccessCount));
+        tl.setResult(tl.TaskResult.Succeeded, tl.loc("Info_SuccessAddingAuth", internalFeedSuccessCount, externalFeedSuccessCount));
```

Comment:
Is setting the task result to success actually necessary? I thought if it just fell through main the task results in a success.

Reply:
You're right it's not! I changed it before for testing, so I changed it back now!